gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManagerUtils.java

```diff
@@ -366,4 +403,62 @@ static List<String> getDistinctUniqueRequesters(String serializedRequesters) {
       throw new RuntimeException("Could not process requesters due to ", e);
     }
   }
+  public static void submitEventsAndSetStatus(Dag<JobExecutionPlan> dag, Optional<EventSubmitter> eventSubmitter) {
+    if (eventSubmitter.isPresent()) {
+      for (DagNode<JobExecutionPlan> dagNode : dag.getNodes()) {
+        JobExecutionPlan jobExecutionPlan = DagManagerUtils.getJobExecutionPlan(dagNode);
+        Map<String, String> jobMetadata = TimingEventUtils.getJobMetadata(Maps.newHashMap(), jobExecutionPlan);
+        eventSubmitter.get().getTimingEvent(TimingEvent.LauncherTimings.JOB_PENDING).stop(jobMetadata);
+        jobExecutionPlan.setExecutionStatus(PENDING);
```

Comment:
why do we only submit pending status events? can we codify that into the method java doc?

Reply:
changed method name, this should clarify. btw, this is an existing code.